master/buildbot/process/buildrequestdistributor.py

```diff
@@ -864,7 +889,30 @@ def nonecmp(a,b):
         defer.returnValue(rv)
 
     @defer.inlineCallbacks
-    def _sortBuilders(self, buildernames):
+    def _globalPrioritySorter(self, master, builders, queue=None):
+        timer = metrics.Timer("BuildRequestDistributor._globalPrioritySorter()")
+        timer.start()
+
+        def findPrioritizedBuildRequest(bldr):
+            d = defer.maybeDeferred(lambda :
+                    bldr.getPrioritizedBuildRequest(queue=queue))
+            d.addCallback(lambda br: (br, bldr))
+            return d
+
+        builRequesPriorityPerBuilder = yield defer.gatherResults(
+                [findPrioritizedBuildRequest(bldr) for bldr in builders])
+
+        priorityBuilders = sorted(builRequesPriorityPerBuilder,
+                                  key=lambda (br, priority_bldr): (-br["priority"], br["submitted_at"])
+                                  if br else (True, True))
+
+        rv = [sorted_bldr[1] for sorted_bldr in priorityBuilders if sorted_bldr[0] is not None]
+        timer.stop()
```

Comment:
Please log this out :) with an easy identifier then we can pick it up in logstash